### ISSUE TITLE
Dice parsing and adjusting in the presence of both absolute dice and sw / thr doesn't always work correctly

### ISSUE THREAD

**Issue by celondon** [richardwilkes/gcs#128]
![image](https://user-images.githubusercontent.com/9532663/63960627-f358dd00-ca5c-11e9-84ce-db11d3c47086.png)

Ranks of Innate Attack are incorrectly multiplying base damage when the ST-Based Attack modifier is applied.

**Comment by richardwilkes** [comment]
Actually, multiplying the damage by the level is the expected behavior for this. There is an issue, though, that the parser can't deal with additional *dice* of damage in addition to something like sw or thr in the damage specification. The best approach here is to ensure the extra dice of damage come first, so list `1d+thr cr` instead. Even this doesn't work well, though, when the resolved sw or thr value isn't a whole dice value (i.e. has a plus/minus). In addition, it won't combine the dice together, either, so for a value of thr = 1d, you'd end up with 8d+8d in the example above.

Nonetheless, for now, the best approach for those entries that try to combine specific dice with sw/thr is to just calculate it by hand and put the correct value in. For example, in the above, you could list the damage as `3d-1 cr` instead. Obviously, this has the downside of needing to be updated by hand should your ST score change.

**Comment by celondon** [comment]
1 Rank with the notation 1d+thr cr gives with a 20 STR gives the following value damage value:
1d+3d-1 -- the 1d is listed separately and, somehow, added to the base thrust damage. It should be 1d+2d-1.

Increasing it to rank 2, the damage increases it to 2d+6d-1 and should be 2d+4d-1.

**Comment by richardwilkes** [comment]
I just verified that the current version of the code does this correctly, **IF** the bonus dice are first in the expression, i.e. `1d+thr cr`. It then resolves to `2d+4d-1 cr` for level 2 and `8d+16d-1 cr` for level 8.

If you then add Brawling at a level such that your RSL is DX+2 or better, then you get an additional point per die... however, that again interacts badly with this notation, regardless of how it is configured, as the parser can't distinguish things properly, so you end up getting the wrong number increased. I don't currently know how to fix this short of a complete rewrite of the parser and am unlikely to try and find a solution anytime soon, as the combination of explicit dice + swing or thrust is incredibly rare in the data files. As I stated before, it is probably better for you to just list the actual pre-calculated damage for the time being.
